oslabs-beta/apollo11
# Fix messaging so that it is tab aware

**Feature**:
Messaging

**Description**:
Fix the message passing so that it is tab aware.
Previously the listeners were accepting all messages from all senders.  This creates issues when there are multiple Apollo11 devtool panels open or multiple Apollo clients open in multiple tabs.
Now the Apollo11 panel will only accept messages from the same tab on which it is open and inspecting traffic.
If it is the same `tabId`, it will process the message.
If it is not the same `tabId`, it will simply ignore it.

**How to test**:
Load the extension and test the following scenarios.
1) Single panel open, single test website open
2) Single panel open, multiple test websites open
3) Multiple panels open, single test website open
4) Multiple panels open, multiple test websites open

In each scenario, perform actions on the test website(s) that would generate GraphQL events.
Confirm that the URI matches the test website that is being inspected.
Confirm that the Event Log contains only the events from the currently inspected tab.
